File: origin/YuGiOhCard/Module/PackItem.m
#import "PackItem.h"

#pragma mark - package cards
@implementation PackageCards

-(PackageCards *) init {
    self = [super init];
    if (self) {
        self.cards = [[NSMutableArray alloc] init];
    }
    return self;
}

@end

#pragma mark - package detail

@implementation PackageDetail

-(PackageDetail *) initWithId:(NSString *)packid packName:(NSString *)packName {
    self = [super init];
    if (self) {
        self.packId = packid;
        self.packName = packName;
    }
    return self;
}

@end

#pragma mark - package item

@implementation PackItem

-(PackItem *) init {
    self = [super init];
    if (self) {
        self.packages = [[NSMutableArray alloc] init];
    }
    return self;
}

-(void) addPackage:(PackageDetail *)item {
    [self.packages addObject:item];
}

@end

File: swift/YuGiOhCard/Module/PackItem.swift
import UIKit


class PackageCards: NSObject {
    var name = ""
    var cards = NSMutableArray()
}


class PackageDetail: NSObject {

    var packId = ""
    var packName = ""
    
    init(packId: String, packageName: String) {
        self.packId = packId
        self.packName = packageName
    }
}


class PackItem: NSObject {
    var serial = ""
    var packages = NSMutableArray()

    func addPackage(item: PackageDetail) {
        self.packages.addObject(item)
    }
}
